src/Chainweb/RestAPI/Checkpoint.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | An endpoint for making RocksDB checkpoints. Used for backups and 
+-- to synchronize databases between nodes.
+
+module Chainweb.RestAPI.Checkpoint
+    ( CheckpointApi
+    , someCheckpointApi
+    , someCheckpointServer
+    ) where
+
+import Control.Monad.IO.Class
+import Data.Proxy
+import Data.Text (Text)
+import Servant
+
+import Chainweb.RestAPI.Utils
+
+type CheckpointApi = "make-checkpoint" :> Get '[PlainText] Text
```

Comment:
That requires we somehow deal with any other checkpoint that may already have that name, I think. We were going to have another service handle that, which may make more sense because it can kill rsync and restart any downloads.